crates/router/src/core/recon.rs

```diff
@@ -0,0 +1,203 @@
+use api_models::recon as recon_api;
+use common_utils::ext_traits::AsyncExt;
+use error_stack::ResultExt;
+use masking::{ExposeInterface, PeekInterface, Secret};
+
+use crate::{
+    core::errors::{self, RouterResponse, StorageErrorExt, UserErrors},
+    services::{
+        api as service_api,
+        authentication::{AuthToken, UserFromToken},
+        email::types as email_types,
+    },
+    types::{
+        api::{self as api_types, enums},
+        domain::{UserEmail, UserName},
+        storage,
+        transformers::ForeignTryFrom,
+    },
+    SessionState,
+};
+
+pub async fn send_recon_request(
+    state: SessionState,
+    user: UserFromToken,
+) -> RouterResponse<recon_api::ReconStatusResponse> {
+    let global_db = &*state.global_store;
+    let db = &*state.store;
+    let key_manager_state = &(&state).into();
+    let merchant_id = user.merchant_id;
+
+    let key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            key_manager_state,
+            &merchant_id,
+            &db.get_master_key().to_vec().into(),
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to fetch merchant's key store")?;
+
+    let merchant_account = db
```

Comment:
Why we reading merchant_account here? Merchant Account should come from auth layer/middleware. How these api are authenticated?

Reply:
This is JWT auth, we don't get merchant account from the auth layer here